File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/item/legends/DistanceTraveled.java
package generations.gg.generations.core.generationscore.common.world.item.legends;

import net.minecraft.server.level.ServerPlayer;
import net.minecraft.world.item.ItemStack;

public interface DistanceTraveled {
    void incrementDistance(ServerPlayer player, ItemStack stack, double distance);
    double getDistance(ItemStack stack);
    void setDistance(ItemStack stack, double distance);
    double getMaxDistance();
    double remainingNeededDistance(ItemStack itemInHand);
}

File: src/common/main/kotlin/generations/gg/generations/core/generationscore/common/world/item/legends/DistanceTraveled.kt
package generations.gg.generations.core.generationscore.common.world.item.legends

import net.minecraft.server.level.ServerPlayer
import net.minecraft.world.item.ItemStack

interface DistanceTraveled {
    fun incrementDistance(player: ServerPlayer, stack: ItemStack, distance: Double)
    fun getDistance(stack: ItemStack): Double
    fun setDistance(stack: ItemStack, distance: Double)
    val maxDistance: Double

    fun remainingNeededDistance(itemInHand: ItemStack): Double
}
